client/components/signup-form/index.jsx

```diff
@@ -55,6 +57,27 @@ export default React.createClass( {
 		}
 	},
 
+	autoFillUsername( form ) {
+		let siteName = getValueFromProgressStore( {
```

Comment:
`let` should be `const` here because we only assign to `siteName` once.

Also, good idea to use a single object in the function signature for `getValueFromProgressStore` and `mergeFormWithValue`. This is much more readable than `getValueFromProgressStore( 'site', 'site', signupProgressStore );`